src/test/java/com/suse/salt/netapi/calls/modules/GitTest.java

```diff
@@ -0,0 +1,177 @@
+
+    static final String JSON_ADD_ERROR_RESPONSE = ClientUtils.streamToString(
+            SaltUtilTest.class.getResourceAsStream("/modules/git/git_add_error.json"));
+
+    static final String JSON_COMMIT_RESPONSE = ClientUtils.streamToString(
+            SaltUtilTest.class.getResourceAsStream("/modules/git/git_commit.json"));
+
+    static final String JSON_BRANCH_RESPONSE = ClientUtils.streamToString(
+            SaltUtilTest.class.getResourceAsStream("/modules/git/git_branch.json"));
+
+    static final AuthMethod AUTH = new AuthMethod(new Token());
+
+    private SaltClient client;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(MOCK_HTTP_PORT);
+
+    @Before
+    public void init() {
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        client = new SaltClient(uri, new HttpAsyncClientImpl(TestUtils.defaultClient()));
+    }
+
+    @Test
+    public void testStatus() {
+        // First we get the call to use in the tests
+        LocalCall<GitResult> call = Git.status("/dev", Optional.empty());
+        assertEquals("git.status", call.getPayload().get("fun"));
+
+        // Test with a successful response
+        mockOkResponseWith(JSON_STATUS_RESPONSE);
+
+        Map<String, Result<GitResult>> response =
+                call.callSync(client, new MinionList("myminion"), AUTH)
+                        .toCompletableFuture().join();
+
+        assertNotNull(response.get("myminion"));
+
+        GitResult output = response.get("myminion").result().get();
+        assertTrue(output.getModified().size() == 2);
+        assertEquals("src/main/java/com/suse/salt/netapi/client/SaltClient.java," +
+                "src/test/java/com/suse/salt/netapi/examples/Calls.java",
+                output.getModified().stream().collect(Collectors.joining(",")));
+    }
+
+    @Test
+    public void testAddSuccess() {
+        // First we get the call to use in the tests
+        LocalCall<String> call = Git.add("/dev", "add_test_file.txt", "", "", Optional.empty());
+        assertEquals("git.add", call.getPayload().get("fun"));
+
+        // Test with a successful response
+        mockOkResponseWith(JSON_ADD_SUCCESS_RESPONSE);
+
+        Map<String, Result<String>> response =
+                call.callSync(client, new MinionList("myminion"), AUTH)
+                        .toCompletableFuture().join();
+
+        assertNotNull(response.get("myminion"));
+
+        String output = response.get("myminion").result().get();
+        assertEquals("add 'add_test_file.txt'", output);
+    }
+
+    @Test
+    public void testAddError() {
+        // First we get the call to use in the tests
+        LocalCall<String> call = Git.add("/dev", "add_test_file.txt", "", "", Optional.empty());
+        assertEquals("git.add", call.getPayload().get("fun"));
+
+        // Test with a successful response
+        mockOkResponseWith(JSON_ADD_ERROR_RESPONSE);
```

Comment:
This comment doesn't seem to be accurate here.

Reply:
Hello @renner, thanks for your comments, I just uploaded a fix on the comment to match the received response.